### PR TITLE
Convert text to speech 

### PR DESCRIPTION
**Pull request by bezlin6mechminerz**
Link to the tested workshop:- https://hackclub.slack.com/files/UJMRP65K9/F01BCC0UJ2Z/tts.mp4
This is a workshop to convert text to speech using python.

### CODE REVIEW

**Review by sampoder — CHANGES_REQUESTED** [review]
Could we extend this workshop out? It's quite short.

Also my most recent review on #1316 is relevant.

**Comment by bezlin6mechminerz** [comment]
Ok

**Comment by bezlin6mechminerz** [comment]
I will do that soon

File: workshops/Convert text to speech/README.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+## Final Output.
+
+<img width="1440" alt="Screenshot 2020-09-19 at 12 43 28 PM" src="https://cloud-amkazzxks.vercel.app/93708256-6181ad80-fb52-11ea-9c6b-f1d7534c5520.png">
+
+So we will be converting text to speech using python today.
+
+View a [live demo](https://repl.it/@bezlin/tts#voice.mp3)
+
+View the [final code](https://repl.it/@bezlin/tts#main.py) .
```
Comment (lachlanjc) [inline workshops/Convert text to speech/README.md:15]:
```suggestion
View the [final code](https://repl.it/@bezlin/tts#main.py)
```

File: workshops/Convert text to speech/README.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
+---
+name:'Convert Text to speech.'
+description:'We will convert text to speech using python.'
+author: '@bezlin6mechminerz'
+---
+
+## Final Output.
+
+<img width="1440" alt="Screenshot 2020-09-19 at 12 43 28 PM" src="https://cloud-amkazzxks.vercel.app/93708256-6181ad80-fb52-11ea-9c6b-f1d7534c5520.png">
```
Comment (lachlanjc) [inline workshops/Convert text to speech/README.md:9]: Missing descriptive alt text, & use Markdown syntax for image

File: workshops/Convert text to speech/README.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+For getting this module you need to install the gtts package. Don't worry if you are using repl.it.In repl.it it will be automatically installed when you run the code.
+But if you are using another platform you need to install the packages via terminal
+
+To install the gTTS API, open terminal/cmd and write
+
+`pip install gTTS`
+
+Now we are all set to write the code.
+
+## Lets code
```
Comment (lachlanjc) [inline workshops/Convert text to speech/README.md:48]: Punctuation

File: workshops/Convert text to speech/README.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+To get started, go to [repl](https://repl.it/languages/Python). Your coding environment will be ready in a few seconds!
+<img width="1440" alt="Screenshot 2020-09-18 at 5 39 06 PM" src="https://cloud-o8q46nkko.vercel.app/screenshot_2020-09-25_at_9.23.06_pm.png">
+
+## Step 2: Get started to code
+
+To convert text to speech we need to import a new Library/API. There are several APIs available to convert text to speech in python. One such APIs is the Google Text to Speech API commonly known as the gTTS API. gTTS is a very easy-to-use tool that converts the text entered, into audio which can be saved as an mp3 file.
+The gTTS API supports several languages including English, Hindi, Tamil, French, German, and many more. The speech can be delivered in any one of the two available audio speeds, fast or slow. However, as of the latest update, it is not possible to change the voice of the generated audio.
+
+For getting this module you need to install the gtts package. Don't worry if you are using repl.it.In repl.it it will be automatically installed when you run the code.
```
Comment (lachlanjc) [inline workshops/Convert text to speech/README.md:39]: You should assume folks are using replit here. The grammar is off here/on the next line

File: workshops/Convert text to speech/README.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+But if you are using another platform you need to install the packages via terminal
+
+To install the gTTS API, open terminal/cmd and write
+
+`pip install gTTS`
```
Comment (lachlanjc) [inline workshops/Convert text to speech/README.md:44]: Unclear if you need to do this on replit

File: workshops/Convert text to speech/README.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+![ezgif com-gif-to-mp4](https://cloud-5m2nwfs8r.vercel.app/ezgif.com-video-to-gif-4.gif)
+
+## Play it
+
+You may have noticed a new file generated after the code has finished running. If not see the folder menu where your main.py file is present you will notice the .mp3 file there. That's the generated file click on it then a small player appears then play it.
+
+## Hacking away!
+
+Now you know how to convert. Try it with other languages and have fun. Try to make a help part in your codes which says the whole help as voice .its pretty cool right?. Elaborate on it and bring more features.
```
Comment (lachlanjc) [inline workshops/Convert text to speech/README.md:104]: Punctuation/grammar off here

File: workshops/Convert text to speech/README.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+print("Converted")
+```
+
+We have converted the text to speech and saved it but for fun lets also print converted at the end of the code. So we can make sure the conversion has completed.
+
+<img width="1440" alt="Screenshot 2020-09-18 at 5 39 06 PM" src="https://cloud-94jt398a9.vercel.app/0screenshot_2020-10-10_at_11.07.31_am.png">
+
+## Run it
+
+![ezgif com-gif-to-mp4](https://cloud-5m2nwfs8r.vercel.app/ezgif.com-video-to-gif-4.gif)
```
Comment (lachlanjc) [inline workshops/Convert text to speech/README.md:96]: Alt text on this & above

File: workshops/Convert text to speech/README.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+myobj = gTTS(text=mytext, lang=language, slow=False)
+```
+
+Here in this line we are declaring another variable 'myobg' and we are giving it the value as calling the module we imported and the gtts is a function here. It takes 3 arguments one is the text. Here we have declared our text in a variable before we need to give it here. Then the language. We also assigned a variable and collected which language we want we also need to specify the variable name here. Then comes the speed you want slow = False means the voice will be faster. Slow=True means the voice will be slow.
+
+```python
+myobj.save("voice.mp3")
+```
+
+So before the above code, the text is converted to the voice but we need to save it right? To save it, the gtts module itself has a save function. the variable we converted to voice .save that's it And also give a name for the file which will be saved here it is voice.mp3.Hope you got it.
```
Comment (lachlanjc) [inline workshops/Convert text to speech/README.md:84]: “Hope you got it”—it’s really important that we use a welcoming tone for beginners

**Review by lachlanjc — CHANGES_REQUESTED** [review]
I’d love to have this workshop, but currently this is just not complete enough & doesn’t create a compelling enough final product. Workshops should be focused on making an end product beginners will be interested in, not on the technology they use to achieve this. This workshop is just about the tech, & barely mentions the usecases at the end. You could solve this by making it a website with an input that generates the file on the backend & plays it, or making a barebones Siri-like thing that responds to you using this technology in some way, etc, but just using the module to generate the file inside code isn’t enough. Additionally, this workshop is missing alt text on images, there are many grammar/punctuation errors (I flagged a few but they are throughout), & the instructions for repl.it users are a bit confusing. I’m closing for now, but feel free to re-open if you’d like to address these for the bounty. Thanks so much.

**Comment by bezlin6mechminerz** [comment]
ok

**Pull request closed without merging**